src/common/modify-settings-modal/index.tsx

```diff
@@ -358,24 +365,42 @@ const ModifySettingsModal = ({ position, open, onCancel }: ModifySettingsModalPr
     variant?: 'text' | 'outlined' | 'contained';
   }[] = [];
 
+  let Component;
+
   if (needsToApprove) {
-    actions = [
-      {
-        color: 'primary',
-        variant: 'contained',
-        label: (
+    Component = (
+      <SplitButton
+        onClick={() => handleApproveToken()}
+        text={
           <FormattedMessage
             description="Allow us to use your coin"
-            defaultMessage="Approve {token}"
+            defaultMessage="Approve Max {token}"
             values={{
-              token: fromToUse.symbol,
+              token: (from && from.symbol) || '',
```

Comment:
this should be `fromToUse` since it will always show as ETH for protocol tokens for example